apps/chat-e2e/src/assertions/folderAssertion.ts

```diff
@@ -418,4 +424,51 @@ export class FolderAssertion<T extends Folders> {
       .soft(actualFoldersCount, ExpectedMessages.foldersCountIsValid)
       .toBe(expectedCount);
   }
+
+  public async assertFoldersSorting(sorting: Sorting) {
+    const allFolderNames = await this.folder.getFolderNames();
+    const sortedNames = allFolderNames.slice().sort((a, b) => {
+      const nameA = a.toLowerCase();
+      const nameB = b.toLowerCase();
+      if (nameA < nameB) {
+        return -1;
+      }
+      if (nameA > nameB) {
+        return 1;
+      }
+      return 0;
+    });
+    const expectedOrder =
+      sorting === 'asc' ? sortedNames : sortedNames.reverse();
+    expect
+      .soft(allFolderNames, ExpectedMessages.elementsOrderIsCorrect)
+      .toEqual(expectedOrder);
+  }
+
+  public async assertFolderSelectedState(
+    folder: TreeEntity,
+    isSelected: boolean,
+  ) {
+    await expect
+      .soft(
+        this.folder.getFolderByName(folder.name, folder.index),
+        ExpectedMessages.folderIsHighlighted,
+      )
+      .toHaveAttribute(Attributes.ariaSelected, String(isSelected));
+  }
+
+  public async assertSearchResult(searchFolderPath: string) {
```

Comment:
I think that the method's overall logic could be clarified with additional comments.

Reply:
added comments